### PR TITLE
add liveness probe to test (example CNF) pod

### PR DESCRIPTION
**Pull request by hamadise**

### CODE REVIEW

**Comment by hamadise** [comment]
I found one main raison why my workflow was failing. 
our github CI always picks the image from quay registry, it's not using the newly build image to run the tests. This is causing CI to send liveness probe to old image. 
To verify this, I actually modified image source to pull from my 'private repo' and it did work.
@jun-rh should we merge this and raise a bug to track the issue ?
Otherwise, I need to fix that bug before this can be merged. It's also worth noting that once this merged (even with the failure), it will work right afterward.

File: test-target/local-pod-under-test.yaml [hunk 1]
```diff
@@ -23,12 +23,11 @@ spec:
...
       automountServiceAccountToken: false
       containers:
-        - command: [ "/bin/bash", "-c", "echo 'logs' && tail -f /dev/null" ]
-          image: quay.io/testnetworkfunction/cnf-test-partner:latest
+        - image: quay.io/testnetworkfunction/cnf-test-partner:latest 
           imagePullPolicy: IfNotPresent
           name: test
           ports:
-            - name: "testport"
+            - name: "probeport"
```
Comment (jun-rh) [inline test-target/local-pod-under-test.yaml:30]: Just FYI, verizon wants the port name to be "protocol-name" and we may add that test soon

Comment (hamadise) [reply test-target/local-pod-under-test.yaml:30]: do you you which section this recommendation is mentioned ? is it on 1.3 document ?

**Comment by jun-rh** [comment]
> I found one main raison why my workflow was failing. our github CI always picks the image from quay registry, it's not using the newly build image to run the tests. This is causing CI to send liveness probe to old image. To verify this, I actually modified image source to pull from my 'private repo' and it did work. @jun-rh should we merge this and raise a bug to track the issue ? Otherwise, I need to fix that bug before this can be merged. It's also worth noting that once this merged (even with the failure), it will work right afterward. This is a good run https://github.com/test-network-function/cnf-certification-test-partner/runs/4620343925?check_suite_focus=true with this commit [39eef99](https://github.com/test-network-function/cnf-certification-test-partner/commit/39eef99e9bc6a0ee001ad8e0a77423136a822a0c)

I am ok with merging this. If it doesn't work as expected, we can always revert. Although, if i understand what you are saying, all it takes is to separate this PR into two? I am sure that Gonzalo would ask you to separate it whether it impacts workflow result or not :)

File: test-target/local-pod-under-test.yaml (hunk 1 above)
Comment (jun-rh) [reply test-target/local-pod-under-test.yaml:30]: Yes, in 1.3, section 19.3

Comment (hamadise) [reply test-target/local-pod-under-test.yaml]: fixed.

**Comment by hamadise** [comment]
> > I found one main raison why my workflow was failing. our github CI always picks the image from quay registry, it's not using the newly build image to run the tests. This is causing CI to send liveness probe to old image. To verify this, I actually modified image source to pull from my 'private repo' and it did work. @jun-rh should we merge this and raise a bug to track the issue ? Otherwise, I need to fix that bug before this can be merged. It's also worth noting that once this merged (even with the failure), it will work right afterward. This is a good run https://github.com/test-network-function/cnf-certification-test-partner/runs/4620343925?check_suite_focus=true with this commit [39eef99](https://github.com/test-network-function/cnf-certification-test-partner/commit/39eef99e9bc6a0ee001ad8e0a77423136a822a0c)
> 
> I am ok with merging this. If it doesn't work as expected, we can always revert. Although, if i understand what you are saying, all it takes is to separate this PR into two? I am sure that Gonzalo would ask you to separate it whether it impacts workflow result or not :)

The fix requires updating  the workflow to use the newly built image. This part of the code is not there yet. It's a separate bug that I did not fix yet.
Let's merge it then tomorrow once Gonzalo takes a look as well. In the meantime, I'll start working on a fix for github CI

**Comment by jun-rh** [comment]
> > > I found one main raison why my workflow was failing. our github CI always picks the image from quay registry, it's not using the newly build image to run the tests. This is causing CI to send liveness probe to old image. To verify this, I actually modified image source to pull from my 'private repo' and it did work. @jun-rh should we merge this and raise a bug to track the issue ? Otherwise, I need to fix that bug before this can be merged. It's also worth noting that once this merged (even with the failure), it will work right afterward. This is a good run https://github.com/test-network-function/cnf-certification-test-partner/runs/4620343925?check_suite_focus=true with this commit [39eef99](https://github.com/test-network-function/cnf-certification-test-partner/commit/39eef99e9bc6a0ee001ad8e0a77423136a822a0c)
> > 
> > 
> > I am ok with merging this. If it doesn't work as expected, we can always revert. Although, if i understand what you are saying, all it takes is to separate this PR into two? I am sure that Gonzalo would ask you to separate it whether it impacts workflow result or not :)
> 
> The fix requires updating the workflow to use the newly built image. This part of the code is not there yet. It's a separate bug that I did not fix yet. Let's merge it then tomorrow once Gonzalo takes a look as well. In the meantime, I'll start working on a fix for github CI

Ok, I misunderstood. Let's merge then.

**Review by jun-rh — APPROVED** [review]

**Pull request merged**